src/ocean/task/TaskPool.d

```diff
@@ -193,7 +153,7 @@ class TaskPool ( TaskT : Task ) : ObjectPool!(Task)
 
             ++count;
 
-            task.terminationHook({
+            task.terminationHookUnique({
```

Comment:
@mihails-strasuns 
There is a problem in one application where the task (main task) gets suspended below (`if (count > 0)  current_task.suspend()`) and the termination hook added here is never executed. 

This is pretty much the scenario I have: 
```
- Scheduler stats when `awaitRunningTasks()`  is called:
stats.task_queue_busy: 0
stats.suspended_queue_busy: 0
stats.worker_fiber_busy: 1    (usually fiber busy is greater than 1, but I found this case interesting to debug)

- the variable `count` in this method goes greater than zero

- [ocean.task.Task] <0X00007F30FB70A580> is suspending itself (main/current task)

- the task <0X00007F30FB70A580> is never resumed
```

Reply:
the number of items in `tasks_iterator` is always equal to `count` so it seems the check `if (!this.isBusy(this.toItem(task)))`  does not kick in

Reply:
That rings a bell in combination with observation in another comment - maybe there is an issue about order of termination hooks that I have missed. Trying to craft a synthetic test case now.